src/librustc_codegen_ssa/lib.rs

```diff
@@ -118,6 +118,29 @@ bitflags::bitflags! {
     }
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct CrateNum(pub u32);
+impl CrateNum {
+    pub fn as_usize(self) -> usize {
+        self.0 as usize
+    }
+}
+impl From<def_id::CrateNum> for CrateNum {
+    fn from(crate_num: def_id::CrateNum) -> Self {
+        CrateNum(crate_num.as_u32())
+    }
+}
+impl Into<def_id::CrateNum> for CrateNum {
+    fn into(self) -> def_id::CrateNum {
+        def_id::CrateNum::from_u32(self.0)
+    }
+}
```

Comment:
Is this actually used?

Reply:
Yes, it is used in src/librustc_codegen_ssa/back/write.rs:995.
`each_linked_rlib_for_lto.push((cnum.into(), path.to_path_buf()));`

Reply:
I think you should use this `CrateNum` for `each_linked_rlib_for_lto` and `exported_symbols` too.

Reply:
Good point, updated.